lib/Jsrt/ChakraDebug.h

```diff
@@ -0,0 +1,567 @@
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The runtime should be active on the current thread and in debug state.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagStopDebugging(
+            _In_ JsRuntimeHandle runtimeHandle,
+            _Out_ void** callbackState);
+
+    /// <summary>
+    ///     Request the runtime to break on next JavaScript statement.
+    /// </summary>
+    /// <param name="runtimeHandle">Runtime to request break.</param>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The runtime should be in debug state. This API can be called from another runtime.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagRequestAsyncBreak(
+            _In_ JsRuntimeHandle runtimeHandle);
+
+    /// <summary>
+    ///     List all breakpoints in the current runtime.
+    /// </summary>
+    /// <param name="breakpoints">Array of breakpoints.</param>
+    /// <remarks>
+    ///     <para>
+    ///     [{
+    ///         "breakpointId" : 1,
+    ///         "scriptId" : 1,
+    ///         "line" : 0,
+    ///         "column" : 62
+    ///     }]
+    ///     </para>
+    /// </remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The current runtime should be in debug state. This API can be called when runtime is at a break or running.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagGetBreakpoints(
+            _Out_ JsValueRef *breakpoints);
+
+    /// <summary>
+    ///     Sets breakpoint in the specified script at give location.
+    /// </summary>
+    /// <param name="scriptId">Id of script from JsDiagGetScripts or JsDiagGetSource to put breakpoint.</param>
+    /// <param name="lineNumber">0 based line number to put breakpoint.</param>
+    /// <param name="columnNumber">0 based column number to put breakpoint.</param>
+    /// <param name="breakPoint">Breakpoint object with id, line and column if success.</param>
+    /// <remarks>
+    ///     <para>
+    ///     {
+    ///         "breakpointId" : 1,
+    ///         "line" : 2,
+    ///         "column" : 4
+    ///     }
+    ///     </para>
+    /// </remarks>
+    /// <returns>
+    ///     The code <c>JsNoError</c> if the operation succeeded, a failure code otherwise.
+    /// </returns>
+    /// <remarks>
+    ///     The current runtime should be in debug state. This API can be called when runtime is at a break or running.
+    /// </remarks>
+    CHAKRA_API
+        JsDiagSetBreakpoint(
+            _In_ unsigned int scriptId,
+            _In_ unsigned int lineNumber,
+            _In_ unsigned int columnNumber,
+            _Out_ JsValueRef *breakPoint);
```

Comment:
done

---
In reply to: [62729231](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62729231) [](ancestors = 62729231)